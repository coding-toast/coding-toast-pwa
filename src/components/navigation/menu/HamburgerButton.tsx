import * as React from 'react'
import HamburgerIcon from '../../../images/hamburger.png'
import Menu from './Menu'
import HamburgerStyles from './HamburgerButton.module.scss'

const HamburgerButton: React.FC = () => {
  const [open, setIsOpen] = React.useState(false)
  return (
    <div>
      <button type="button" className={HamburgerStyles.hamburgerButton} onClick={() => setIsOpen(!open)} onKeyDown={() => setIsOpen(!open)}>
        <img src={HamburgerIcon} className={HamburgerStyles.hamburgerIcon} alt="Open Menu" />
      </button>
      {open ? <Menu /> : ''}
    </div>
  )
}

export default HamburgerButton
