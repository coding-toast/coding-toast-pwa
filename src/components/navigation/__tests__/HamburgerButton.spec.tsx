import React from 'react'
import { shallow } from 'enzyme'
import HamburgerButton from '../menu/HamburgerButton'

describe('HamburgerButton', () => {
  it('will render correctly', () => {
    const component = shallow(<HamburgerButton />)
    expect(component).toMatchSnapshot()
  })

  it('will open correctly', async () => {
    const component = shallow(<HamburgerButton />)

    await component.find('.hamburgerButton').simulate('click')

    expect(component).toMatchSnapshot()
  })
})
