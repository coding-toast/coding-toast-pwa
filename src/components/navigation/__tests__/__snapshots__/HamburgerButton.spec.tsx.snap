// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerButton will open correctly 1`] = `
<div>
  <button
    className="hamburgerButton"
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "background": "none",
        "border": "none",
        "color": "inherit",
        "cursor": "pointer",
        "font": "inherit",
        "outline": "inherit",
        "padding": 0,
      }
    }
    type="button"
  >
    <img
      alt="Open Menu"
      src="test-file-stub"
      style={
        Object {
          "color": "#70F3FF",
          "display": "block",
          "padding": "0",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </button>
  <Menu />
</div>
`;

exports[`HamburgerButton will render correctly 1`] = `
<div>
  <button
    className="hamburgerButton"
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "background": "none",
        "border": "none",
        "color": "inherit",
        "cursor": "pointer",
        "font": "inherit",
        "outline": "inherit",
        "padding": 0,
      }
    }
    type="button"
  >
    <img
      alt="Open Menu"
      src="test-file-stub"
      style={
        Object {
          "color": "#70F3FF",
          "display": "block",
          "padding": "0",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </button>
</div>
`;
